pf/tests/extract_inputs_test.go

```diff
@@ -0,0 +1,928 @@
+package tfbridgetests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	rschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/mapdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hexops/autogold/v2"
+	"github.com/pulumi/pulumi-terraform-bridge/pf/internal/schemashim"
+	pb "github.com/pulumi/pulumi-terraform-bridge/pf/tests/internal/providerbuilder"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExtractInputsFromOutputsPF(t *testing.T) {
+	type testCase struct {
+		name      string
+		props     resource.PropertyMap
+		resSchema rschema.Schema
+		expect    autogold.Value
+	}
+
+	testCases := []testCase{
+		// ATTRIBUTES
+		{
+			name:  "string extracted",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": "bar"}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Optional: true},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{
+				resource.PropertyKey("__defaults"): resource.PropertyValue{
+					V: []resource.PropertyValue{},
+				},
+				resource.PropertyKey("foo"): resource.PropertyValue{V: "bar"},
+			}),
+		},
+		// TODO[pulumi/pulumi-terraform-bridge#2218]: This should not yield values for foo in the inputs.
+		{
+			name:  "string with default not extracted",
+			props: resource.NewPropertyMapFromMap(map[string]interface{}{"foo": "bar"}),
+			resSchema: rschema.Schema{
+				Attributes: map[string]rschema.Attribute{
+					"foo": rschema.StringAttribute{Optional: true, Default: stringdefault.StaticString("bar")},
+				},
+			},
+			expect: autogold.Expect(resource.PropertyMap{
+				resource.PropertyKey("__defaults"): resource.PropertyValue{
+					V: []resource.PropertyValue{},
+				},
+				resource.PropertyKey("foo"): resource.PropertyValue{V: "bar"}, // wrong
+			}),
+		},
```

Comment:
I would recommend trying converting these table based tests to sub-tests like we've done here:
https://github.com/pulumi/pulumi-azure-native/blob/3a4638df1e9b832261333d784e19983dbfb76a6a/provider/pkg/convert/responseBodyToSdkOutputs_test.go#L24-L40

This allows us to use `t.Skip("[REASON]")` with the correct asserts in place rather than asserting on incorrect values. It also has some added usibility benefits like the test failures point to correct line numbers.

Reply:
Thanks for the recommendation - I'll give the sub-tests a try next time.

As for asserting on incorrect values, I don't agree about skipping them - we need to be very intentional about changes here and these are not tested otherwise. The unskipped tests for wrong values helped us discover a regression in the PF bridge which was believe to only affect SDKv2: https://github.com/pulumi/pulumi-terraform-bridge/pull/2181#issuecomment-2244694968

Reply:
Oh I see ... perhaps the comment was misleading - so it's not always an incorrect value - only incorrect in some circumstances? Would you expect this test to fail once you've fixed the bug?

Reply:
Yeah, the test should fail once we fix the bug.